internal/sponge/item/item.go

```diff
@@ -33,6 +34,14 @@ func Upsert(context interface{}, db *db.DB, item *Item) error {
 		return err
 	}
 
+	// If CreatedAt is not set, set it. This allows for CreatedAt to be set for data import.
+	if item.CreatedAt.IsZero() {
```

Comment:
The creation of an id is decoupled from setting CreatedAt in cases where items are being imported as data and the ids are being inferred. Also, there are times when a frontend will want to Create a new item but specify it's ID, such as creating a "singleton" item that can be identified by a given key, such as a default thread, config object, etc...